test/check-existing-system

```diff
@@ -0,0 +1,46 @@
+#!/usr/bin/python3
+#
+# Copyright (C) 2024 Red Hat, Inc.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with this program; If not, see <http://www.gnu.org/licenses/>.
+
+import os
+
+import anacondalib
+from installer import Installer
+from review import Review
+from testlib import test_main  # pylint: disable=import-error
+
+WEBUI_TEST_DIR = os.path.dirname(__file__)
+ROOT_DIR = os.path.dirname(WEBUI_TEST_DIR)
+BOTS_DIR = f'{ROOT_DIR}/bots'
+
+class TestExistingSystemFedora(anacondalib.VirtInstallMachineCase):
```

Comment:
@martinpitt i think I would need some discussion / help on how to convert this to non-destructive.

Let me shortly explain: right now all anaconda tests are running on top of empty qcow2 images and booting the TEST_OS=fedora-rawhide-boot ISO.

In this PR I am however introducing new test cases that run on top already installed images. I started here on top of our rawhide image as a backing_store, but I plan to extend against other images as well, including windows if possible.

My problem is that this test class can't be made easily non-destructive, however all tests inside this class, would nicely share one VM if the non-destructive logic was limited to only take tests inside a specific class.

So I would need here something like @nondestructive_inside_class or similar. 

Does this make sense? Do you have any thoughts on this?

Reply:
Moved out in separate pR https://github.com/rhinstaller/anaconda-webui/pull/329#discussion_r1658934967 
Let's discuss there.